src/chatClient.cpp

```diff
@@ -396,13 +396,28 @@ int Client::importMessages(const char *externalDbPath)
         chatd::Message *newestAppMsg = nullptr;
         chatd::Idx firstIdxToImport = CHATD_IDX_INVALID;    // may not match the idx from app (even for same msgid)
         karere::Id firstMsgidToImport(Id::inval());
+        std::map<karere::Id, uint16_t> messagesUpdated;
 
         std::string query;
         if (!chat.empty())
         {
             newestAppIdx = chat.highnum();
             newestAppMsg = &chat.at(newestAppIdx);
             newestAppMsgid = firstMsgidToImport = newestAppMsg->id();
+            uint32_t tsMissingUpdates = newestAppMsg->ts - CHATD_MAX_EDIT_AGE;
+
+            chatd::Idx messageIdx  = chat.highnum() - 1;
+            while (messageIdx > chat.lownum())
```

Comment:
It will only consider messages loaded in RAM, which is very likely enough for most cases, but it could happen that some messages in DB (not loaded yet) have been updated too.

Also, this block could be avoided in case the app's DB is ahead of NSE's DB, or in case there is a truncate.

Reply:
Fixed